### PR TITLE
Market page screenshots lightbox

### LINKED ISSUE #287: Implement opening screenshots in lightbox

### PR DESCRIPTION
**Pull request by bartaz**
Fixes #287 

Adds previewing screenshots in lightbox to market page

## QA

- ./run or http://snapcraft.io-pr-314.run.demo.haus/
- go to market page of a snap you own
- if it doesn't have screenshots add some (you don't have to upload them)
- click on a 'fullscreen' button next to screenshots
- screenshots should open in lightbox
- you should be able to navigate through screenshots (clicking next/prev buttons)

screen shot 2018-02-16 at 11 52 28

screen shot 2018-02-16 at 11 53 03

### CODE REVIEW

**Comment by tbille** [comment]
Questions:
- Should we adapt the size of the images so that they fit in the size of the screen (see screenshot)? 
- It seems natural to add a double click event on the screenshot to open the lightbox.
- Same for arrows to move between images
- Click outside should quit the image?

Maybe all those events should be discussed and done in another PR

**Example of big image:**
screenshot-2018-2-19 market details for toto

File: static/js/publisher/market/market.js [hunk 1]
```diff
@@ -121,6 +142,8 @@ function initSnapScreenshotsEdit(screenshotsToolbarElId, screenshotsWrapperElId,
       input.click();
     }
 
+
+
```
Comment (tbille) [inline static/js/publisher/market/market.js]: remove lines

File: static/js/publisher/publisher.js [hunk 2]
```diff
@@ -2,4 +2,5 @@ import metrics from './metrics/metrics';
 import { selector } from './metrics/filters';
 import market from './market/market';
 
+
```
Comment (tbille) [inline static/js/publisher/publisher.js]: Remove line

**Comment by Greggless** [comment]
@tbille Good questions, and should probably get this merged and we can make enhancements.

> Should we adapt the size of the images so that they fit in the size of the screen (see screenshot)?
- Yes 

> It seems natural to add a double click event on the screenshot to open the lightbox
- I agree

> Same for arrows to move between images
- What do you mean? 

> Click outside should quit the image?
- Yes clicking outside of the image (on the dark background, should close it) (@matthewpaulthomas, thoughts?)
- We should consider using a similar overlay style that we use on the arrows for the `X` as it may not be noticeable on a dark background (@lyubomir-popov, thoughts?)

# 🌈

**Comment by tbille** [comment]
@Greggless 
> Same for arrows to move between images

I mean using the arrow keys (right and left) of the keyboard to move between images.

**Comment by Lukewh** [comment]
@Greggless @tbille please try again, I've updated this PR thanks!

**Comment by lyubomir-popov** [comment]
@Lukewh @bartaz 

Hi, there are a few differences between this pr and the design ( https://github.com/CanonicalLtd/snapcraft-design/pull/290 ):
• No pagination dots
• Arrows need to be black inside white circles
• close needs to be smaller, white

Not in the design, but nice to have - keyboard control, swiping as in http://idangero.us/swiper/demos/

For reference juju implementation of the same UI - https://jujucharms.com/new -> click on help in top right corner, then click on text link in third card; 
image

**Comment by Lukewh** [comment]
@lyubomir-popov as this design has only just been submitted (an hour ago) and the new design requires significant changes to the code (this lightbox was pulled directly from MAAS) I'm going to:
- Change the arrows to be black inside white circles
- Change the close button to be smaller
The keyboard shortcuts are already implemented.

I'll create a follow up issue to re implement the lightbox in a future iteration: https://github.com/canonical-websites/snapcraft.io/issues/326

**Comment by lyubomir-popov** [comment]
@Lukewh sounds good to me, the close and arrows are the most important changes. 

On the recency of this image - The design was signed off and is in an old issue somewhere, I just noticed it hadn't been put in the code tab and made a pr from that - nothing in it has changed since it was signed off in November.

**Comment by Lukewh** [comment]
QA:

When the lightbox is open you should be able to navigate left and right with the keyboard arrow keys and close the lightbox by pressing escape.
You should also be able to click outside the image to close the lightbox.
Double clicking on an image (when lightbox is closed) should also open the lightbox on that image.

**Pull request merged**